baybe/kernels/base.py

```diff
@@ -19,21 +18,42 @@
 class Kernel(ABC, SerialMixin):
     """Abstract base class for all kernels."""
 
-    lengthscale_prior: Optional[Prior] = field(default=None, kw_only=True)
-    """An optional prior on the kernel lengthscale."""
-
     def to_gpytorch(self, *args, **kwargs):
         """Create the gpytorch representation of the kernel."""
         import gpytorch.kernels
 
+        # Due to the weird way that gpytorch does the inheritance for kernels, it is not
+        # sufficient to just check for the fields of the actual class, but also for the
+        # ones of the basic "Kernel" class and combine them.
         kernel_cls = getattr(gpytorch.kernels, self.__class__.__name__)
-        fields_dict = filter_attributes(object=self, callable_=kernel_cls.__init__)
+        base_cls = getattr(gpytorch.kernels, "Kernel")
+        kernel_fields_dict = filter_attributes(
+            object=self, callable_=kernel_cls.__init__
+        )
+        base_fields_dict = filter_attributes(object=self, callable_=base_cls.__init__)
+        fields_dict = kernel_fields_dict | base_fields_dict
+
+        # Since the args and kwargs passed in this function are kernel-specific, we do
+        # not need to pass them when converting the priors but when converting the
+        # kernels.
+        prior_dict = {
+            key: fields_dict[key].to_gpytorch()
+            for key in fields_dict
+            if isinstance(fields_dict[key], Prior)
+        }
 
-        # If a lengthscale prior was chosen, we manually add it to the dictionary
-        if self.lengthscale_prior is not None:
-            fields_dict["lengthscale_prior"] = self.lengthscale_prior.to_gpytorch()
+        # NOTE: Our coretest actually behave the same if we do not pass *args and
```

Comment:
?? can you elaborate? before merge this needs to be understood

Reply:
Based on a very quick glance I had at the gpytoch code, it seems like the GPyTorch ScaleKernel directly passes its `kwargs` to the base kernel:
![image](https://github.com/emdgroup/baybe/assets/84016733/c3238ab3-1e1c-4963-8fa3-9819b993d685)
This would mean that we do not need to do this, explaining why it does not seem to make a difference. However, I do not know whether this is a behavior special for ScaleKernels or if they use this in other places. Hence the question how to deal with this here.

Reply:
This is one of the things I meant when I said "things are becoming difficult to understand" with the automerge approach. Agree with @Scienfitz that we should be 100% clear of what's going on before merging. Thats is, what stuff can be potentially contained in `*args` and `*kwargs`? I'm already a bit lost at the moment since I didn't write the code myself...

Reply:
Since we agreed to just keep `*args` and `**kwargs` here, I'll resolve.